src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -2515,6 +2517,40 @@ create_exp_macro(Exp, exp)
 create_exp_macro(Exp2, exp2)
 create_exp_macro(Expm1, expm1)
 
+namespace Input {
+
+    static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x,
+            diag::Diagnostics& diagnostics) {
+        ASRUtils::require_impl(x.n_args <= 1,
+            "ASR Verify: Call `input` takes 0 or 1 argument",
+            x.base.base.loc, diagnostics);
+        if (x.n_args == 1) {
+            ASR::ttype_t *type = ASRUtils::expr_type(x.m_args[0]);
+            ASRUtils::require_impl(ASRUtils::is_character(*type),
+                "ASR Verify: Argument to `input` must be of string type",
+                x.base.base.loc, diagnostics);
+        }
+    }
+
+    static inline ASR::asr_t* create_Input(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        if (args.n != 1) {
+            err("input() takes 0 or 1 argument", loc);
+        }
+        if (!ASRUtils::is_character(*ASRUtils::expr_type(args[0]))) {
+            err("Argument `prompt` to input() must be of type 'str'",
+                loc);
+        }
+        std::string dummy_string = "";
+        ASR::ttype_t* return_type = expr_type(StringConstant(dummy_string, character(dummy_string.length())));
+        return ASR::make_IntrinsicScalarFunction_t(al, loc,
+            static_cast<int64_t>(IntrinsicScalarFunctions::Input),
+            args.p, args.n, 0, return_type, nullptr);
+    }
+
+} // namespace Input
```

Comment:
I am not sure if implementing `input()` as an intrinsic registry function is a best approach. But I think we can go ahead with it for now. If we find another approach later (for example using an ASR node), we can refactor it then.